cpp/velox/shuffle/VeloxShuffleWriter.h

```diff
@@ -136,7 +134,7 @@ class VeloxShuffleWriter : public ShuffleWriter {
 
   // Memory Pool used to track memory usage of partition buffers.
   // The actual allocation is delegated to options_.memoryPool.
-  std::unique_ptr<ShuffleMemoryPool> partitionBufferPool_;
+  std::shared_ptr<arrow::MemoryPool> partitionBufferPool_;
```

Comment:
Why do we change to shared_ptr here? Is the pool shared?

Reply:
I see. it's used by ptr.lock(). Add a comment here that
VeloxShuffleWritter owns the ownership actually.